### PR TITLE
Add variables and comments

### PR DESCRIPTION
**Pull request by mh0lt**
Additional comments and variables for:
 
https://github.com/anacrolix/torrent/commit/41fc76adf87d3d640886e4ddf614fc92ce7520e4

### CODE REVIEW

**Review by anacrolix — COMMENTED** [review]
Oops, I forgot to submit this review.

File: webseed-peer.go [hunk 1]
```diff
@@ -117,10 +119,38 @@ start:
 			ws.requesterCond.L.Lock()
 			return false
 		})
-		if restart {
-			goto start
+
+		if !restart {
```
[the remark this answers is not in the document]
Comment (AskAlexSharov) [reply webseed-peer.go:123]: @mh0lt let's break it to 2 PR's?

[the remark this answers is not in the document]
Comment (anacrolix) [reply webseed-peer.go:123]: If you rebase you should find less changes are needed too. Thanks for your efforts.

**Comment by anacrolix** [comment]
I think this is related to https://github.com/anacrolix/torrent/pull/932#issuecomment-2192766727 and the same reasoning applies. Thanks!

**Pull request closed without merging**